js/rack.js

```diff
@@ -31,228 +31,9 @@
-                                var new_x = parseInt(j_item.attr('gs-x'));
-                                var new_y = parseInt(j_item.attr('gs-y'));
-                                if (j_item.attr('gs-w') == 1) {
-                                    new_x = (j_item.attr('gs-x') == 0 ? 1 : 0);
-                                }
-                                dirty = true;
-
-                                // update other side element coordinates
-                                other_side_grid.update(other_side_el[0], {
-                                    'x': new_x,
-                                    'y': new_y
-                                });
-                                dirty = false;
-                            }
-                        }
-                    }).fail(function() {
-                        // reset to old position
-                        dirty = true;
-                        grid.update(item.el, {
-                            'x': x_before_drag,
-                            'y': y_before_drag
-                        });
-                        dirty = false;
-                        displayAjaxMessageAfterRedirect();
-                    });
-                });
-            })
-
-        // store coordinates before start dragging
-            .on('dragstart', function(event) {
-                var element = $(event.target);
-
-                x_before_drag = Number(element.attr('gs-x'));
-                y_before_drag = Number(element.attr('gs-y'));
-
-                // disable qtip
-                element.qtip('hide', true);
-            });
-    });
-
-    $('#viewgraph .cell_add, #viewgraph .grid-stack-item').each(function() {
-        var tipcontent = $(this).find('.tipcontent');
-        if (tipcontent.length) {
-            $(this).qtip({
-                position: {
-                    my: 'left center',
-                    at: 'right center'
-                },
-                content: {
-                    text: tipcontent
-                },
-                style: {
-                    classes: 'qtip-shadow qtip-bootstrap rack_tipcontent'
-                }
-            });
-        }
-    });
-
-    for (var i = grid_rack_units; i >= 1; i--) {
-        // add index number front of each rows
-        $('.indexes').append('<li>' + i + '</li>');
-
-        // append cells for adding new items
-        $('.racks_add').append(
-            '<div class="cell_add"><span class="tipcontent">'+grid_rack_add_tip+'</span></div>'
-        );
-    }
-};
-
-
-var getHpos = function(x, is_half_rack, is_rack_rear) {
-    if (!is_half_rack) {
-        return 0;
-    } else if (x == 0 && !is_rack_rear) {
-        return 1;
-    } else if (x == 0 && is_rack_rear) {
-        return 2;
-    } else if (x == 1 && is_rack_rear) {
-        return 1;
-    } else if (x == 1 && !is_rack_rear) {
-        return 2;
-    }
-};
+$(() => {
+    window.GLPILegacy.loadModule('rack');
+});
```

Comment:
Would this mean keeping it loaded for every page then in case a plugin used the script file for it separately?

Reply:
I just mean that we should keep this file in case any plugin call it directly, but, in the `Html::requireJs()` process, we should be able to find a way to load the module without having to load this "proxy" file.